tcmalloc/internal/percpu.h

```diff
@@ -19,7 +19,8 @@
 
 // PERCPU_RSEQ_SUPPORTED_PLATFORM defines whether or not we have an
 // implementation for the target OS and architecture.
-#if defined(__linux__) && (defined(__x86_64__) || defined(__PPC64__))
+#if defined(__linux__) \
+  && (defined(__x86_64__) || defined(__PPC64__) || defined (__aarch64__))
```

Comment:
On an older Clang (pre https://reviews.llvm.org/rGd2e949eed5b49ca70a8845dfaf7bac6ca40c429d), we won't have `__AARCH64_CMODEL_SMALL__` defined, so we'll later `#error` on its absence in the .S file.

To avoid a build failure, I think we should enable these paths when `defined(__aarch64__) && __AARCH64_CMODEL_SMALL__`.

Reply:
Hi Chris,

I could also condition the check for the memory model with '!defined(__clang_major__) || __clang_major__ >= 9', that would enable the use of RSEQ for older compilers, it would just put the onus on making sure the correct memory model is being used on the developer.

Let me know which one you would prefer?

I've rebased the code too, will post when you pick your favourite solution for this.

Reply:
Checking the compiler version and skipping the check sounds good to me.

For what it's worth, I wouldn't be surprised if the x86 implementation also did not react well to other code models either...